tests/buy_button_test.py

```diff
@@ -0,0 +1,151 @@
+            name="test item2",
+            item_group=2,
+            price=500,
+            limit=1,
+        )
+        item1 = Item (
+            item_type=1,
+            item_group=1,
+            user_id=self.user1_id,
+            count=1,
+            active=False
+        )
+        item2 = Item (
+            item_type=2,
+            item_group=2,
+            user_id=self.user2_id,
+            count=1,
+            active=False
+        )
+        with self.app.app_context():
+            db.session.add(trans1)
+            db.session.add(trans2)
+            db.session.add(store_item1)
+            db.session.add(store_item2)
+            db.session.add(item1)
+            db.session.add(item2)
+            db.session.commit()
+        self.params_item1 = {
+            KEY_ID: 1,
+            KEY_QUANTITY: 1
+        }
+        self.buy_button_bad_funds = {
+            KEY_SUCCESS: False,
+            KEY_MESSAGE: "Insufficient funds"
+        }
+        self.buy_button_bad_item = {
+            KEY_SUCCESS: False,
+            KEY_MESSAGE: "Reached purchase limit"
+        }
+        self.params_item2 = {
+            KEY_ID: 2,
+            KEY_QUANTITY: 1
+        }
+        self.buy_button_params = [
+        {
+            KEY_USER_ID: self.user1_id,
+            KEY_PARAMS: self.params_item1,
+            KEY_AMOUNT: 500,
+            KEY_EXPECTED: { KEY_SUCCESS: True }
+        },
+        {
+            KEY_USER_ID: self.user1_id,
+            KEY_PARAMS: self.params_item2,
+            KEY_AMOUNT: 0,
+            KEY_EXPECTED: { KEY_SUCCESS: True }
+        },
+        {
+            KEY_USER_ID: self.user2_id,
+            KEY_PARAMS: self.params_item1,
+            KEY_AMOUNT: 500,
+            KEY_EXPECTED: self.buy_button_bad_funds
+        },
+        {
+            KEY_USER_ID: self.user3_id,
+            KEY_PARAMS: self.params_item1,
+            KEY_AMOUNT: None,
+            KEY_EXPECTED: self.buy_button_bad_funds
+        },
+        {
+            KEY_USER_ID: self.user2_id,
+            KEY_PARAMS: self.params_item2,
+            KEY_AMOUNT: 500,
+            KEY_EXPECTED: self.buy_button_bad_item
+        }]
+        self.buy_button_error = {
+            KEY_ERROR: "Malformed request"
+        }
+
+    def buy_button_test(self, test_case):
+        with self.app.app_context():
+            with self.client.session_transaction() as sess:
+                sess['user_id'] = test_case[KEY_USER_ID]
+            res = self.client.post('/api/store/buy', data = json.dumps(test_case[KEY_PARAMS]))
+            result = json.loads(res.data.decode("utf-8"))
+            self.assertDictEqual(test_case[KEY_EXPECTED], result)
+            self.assertEqual(
+                test_case[KEY_AMOUNT],
+                db.session.query(func.sum(Transaction.ticket_amount))
+                .filter(Transaction.user_id == test_case[KEY_USER_ID])
+                .scalar()
+            )
+
+    def test_buy_button_success(self):
```

Comment:
here too. We aren't really testing buy button. Applies to all other method names.